docs/tutorial/miscellany.rst

```diff
@@ -276,8 +276,9 @@ type ``Int`` will be provided. Otherwise, it will provide the type
     strToType _ = Nat
 
     fromFile : String -> IO (Provider Type)
-    fromFile fname = do str <- readFile fname
-                        return (Provide (strToType (trim str)))
+    fromFile fname = do Right str <- readFile fname
+		| Left err => pure (Provide Void)
+	pure (Provide (strToType (trim str)))
```

Comment:
I think this would look better as:

```idris 
 fromFile fname = do Right str <- readFile fname
                                      | Left err => pure (Provide Void)
                                  pure (Provide (strToType (trim str)))
```